include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -4343,12 +4359,16 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
     this->scratch_data_array->begin() + n_components * dofs_per_component;
   gradients_quad = this->scratch_data_array->begin() +
                    n_components * (dofs_per_component + n_quadrature_points);
-  hessians_quad =
+  gradients_from_hessians_quad =
     this->scratch_data_array->begin() +
     n_components * (dofs_per_component + (dim + 1) * n_quadrature_points);
-  this->scratch_data =
-    this->scratch_data_array->begin() + n_components_ * dofs_per_component +
-    (n_components_ * ((dim * (dim + 1)) / 2 + dim + 1) * n_quadrature_points);
+  hessians_quad =
+    this->scratch_data_array->begin() +
+    n_components * (dofs_per_component + (2 * dim + 1) * n_quadrature_points);
+  this->scratch_data = this->scratch_data_array->begin() +
+                       n_components_ * dofs_per_component +
+                       (n_components_ * ((dim * (dim + 1)) / 2 + 2 * dim + 1) *
+                        n_quadrature_points);
```

Comment:
Not sure, but don't we only want the memory if MF was actually set up with Hessian enabled?

Reply:
This is what we did before. We maybe waste 4 or 5 instructions here to get the intermediate pointers, against an `if` and some bookkeeping to check if we had hessians filled in `MappingInfo`, so I think the code is cleaner this way. The local arrays are small enough to not really eat much into the memory of a global finite element computation, and in case we have no hessians we will never touch them (and hence not waste cache memory).

Reply:
We could move it to the end of `scratch_data_array` and disable it in case the flag `UpdateFlags::update_jacobian_grads` is not set in `mapping_data->update_flags_inner_faces`. This can also be done in a follow up if we think it is necessary.